tests/test_mathematics.py

```diff
@@ -86,3 +88,37 @@ class TestTranspose(test_tools.TestTranspose):
     specialisations = [
         pytest.param(dense.transpose_cupydense, CuPyDense, CuPyDense),
     ]
+
+
+class TestPow(test_tools._GenericOpMixin):
+
+    # This should be part of QuTiP and only inherited here
+    # TODO: Add it to QuTiP and inherit here
+
+    def op_numpy(self, matrix, scalar):
+
+        return np.linalg.matrix_power(matrix, scalar)
+
+    shapes = [
+        (pytest.param((1, 1),),),
+        (pytest.param((5, 5),),),
+        (pytest.param((10, 10),),),
+    ]
```

Comment:
We should add tests for the exception case too -- i.e. `bad_shapes = [...]` and the suitable error test method.